internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -167,47 +167,114 @@ func (cf CloudFormation) describeStack(describeStackInput *cloudformation.Descri
 func (cf CloudFormation) describeStackWithClient(describeStackInput *cloudformation.DescribeStacksInput,
 	client cloudformationiface.CloudFormationAPI) (*cloudformation.Stack, error) {
 	describeStackOutput, err := client.DescribeStacks(describeStackInput)
-	if err != nil {
+	stackNotFoundError := stackDoesNotExist(err)
+	if err != nil && !stackNotFoundError {
 		return nil, err
 	}
 
-	if len(describeStackOutput.Stacks) == 0 {
-		return nil, fmt.Errorf("failed to find a stack named %s", *describeStackInput.StackName)
+	if stackNotFoundError || len(describeStackOutput.Stacks) == 0 {
+		return nil, &ErrStackNotFound{stackName: *describeStackInput.StackName}
 	}
 
 	return describeStackOutput.Stacks[0], nil
 }
+
+// create will only spin up a stack if none exists or if a stack exists
+// but requires cleanup (meaning we failed to created it before). With
+// stacks that are failed to be created, you have to delete them if you
+// want to update them. In this case, we'll delete the failed stack
+// and then try creating it again.
+// If a stack already exists in another state, we'll return an ErrStackAlreadyExists
+// error.
 func (cf CloudFormation) create(stackConfig stackConfiguration) error {
-	return cf.deploy(stackConfig, cloudformation.ChangeSetTypeCreate)
+	describeStackInput := &cloudformation.DescribeStacksInput{StackName: aws.String(stackConfig.StackName())}
+	existingStack, err := cf.describeStack(describeStackInput)
+	// Create the stack if it doesn't already exists.
+	if err != nil {
+
+		var stackNotFound *ErrStackNotFound
+		if !errors.As(err, &stackNotFound) {
+			return fmt.Errorf("checking to see if a stack already exists: %w", err)
```

Comment:
What do you think about removing this wrapped error? When I read it I thought a follow-up action was going to happen because of "checking to see".

Since the error comes from a private function I'd rather remove the wrapping here and move it inside `describeStack`.